### ISSUE TITLE
Bug Report: `Box::from_raw` does not trigger segmentation fault in VSCode runnable

### ISSUE THREAD

**Issue by Nahida7** [rust-lang/rust-analyzer#19470]
**rust-analyzer version**: (eg. output of "rust-analyzer: Show RA Version" command, accessible in VSCode via <kbd>Ctrl/⌘</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>)

rust-analyzer version: 0.3.2353-standalone (37acea8052 2025-03-23) 

**rustc version**: (eg. output of `rustc -V`)

rustc 1.85.1 (4eb161250 2025-03-15)

**editor or extension**: (eg. VSCode, Vim, Emacs, etc. For VSCode users, specify your extension version; for users of other editors, provide the distribution if applicable)

VSCode (rust-lang.rust-analyzer 0.3.2353)

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

**repository link (if public, optional)**: (eg. [rust-analyzer](https://github.com/rust-lang/rust-analyzer))

**code snippet to reproduce**: 

```rust
fn main() {
    // try to create a Box from a invalid pointer
    let _ = unsafe {
        let mut ptr: *mut i32 = std::ptr::null_mut();
        ptr = ptr.add(10usize);
        Box::from_raw(ptr);
    };
}
```

**Observed behavior:**

When running `cargo run` from the terminal, the program correctly triggers a segmentation fault.

![Image](https://github.com/user-attachments/assets/9c731c39-c939-4a8c-a204-ca732609774f)

However, when running the same code using VSCode’s "Run" button (runnable feature), it does not trigger a segmentation fault.

![Image](https://github.com/user-attachments/assets/5d7b0a22-7437-490a-ba48-1b62ce873aa2)

**Expected behavior:**

The behavior should be consistent: running the code in VSCode should also trigger a segmentation fault, just like when using cargo run in the terminal.

**Additional context:**

It seems that the VSCode runnable feature might be handling process crashes differently. Would appreciate insights on whether this is expected behavior or a bug.

**Comment by Veykril** [comment]
I mean, this is undefined behavior so I don't think it makes sense to try to assume behavior about this. The environment between running in the terminal and running through VSCode can differ which can affect this very easily.

**Comment by ChayimFriedman2** [comment]
As @Veykril said, this is UB, you should not do this, and it's (not) working as expected. Closing as there is no bug here.
